muzzle/src/main/java/io/opentelemetry/javaagent/tooling/HelperInjector.java

```diff
@@ -172,7 +172,8 @@ public DynamicType.Builder<?> transform(
     injectedClassLoaders.computeIfAbsent(
         maskNullClassLoader(classLoader),
         cl -> {
-          List<HelperClassDefinition> helpers = helperClassesGenerator.apply(cl);
+          List<HelperClassDefinition> helpers =
+              helperClassesGenerator.apply(isBootClassLoader(cl) ? null : cl);
```

Comment:
perhaps add a method `unmaskNullClassLoader` to make it clear that this here reverses the previous call to `maskNullClassLoader`

Reply:
Good idea, I've added it in https://github.com/open-telemetry/opentelemetry-java-instrumentation/pull/12136/commits/e5bfd05b8e63b29603853e4ce73e1dde6b1bcb0f